src/lib.rs

```diff
@@ -205,15 +206,20 @@ impl Header {
     }
 }
 
+pub struct Level<'a> {
+    pub bytes: &'a [u8],
+    pub uncompressed_length_bytes: u64,
+}
+
 #[derive(Debug, Eq, PartialEq, Copy, Clone)]
-struct LevelIndex {
-    offset: u64,
-    length_bytes: u64,
-    uncompressed_length_bytes: u64,
+pub struct LevelIndex {
+    pub offset: u64,
+    pub length_bytes: u64,
+    pub uncompressed_length_bytes: u64,
 }
 
 impl LevelIndex {
-    const LENGTH: usize = 24;
+    pub const LENGTH: usize = 24;
 
     pub fn from_bytes(data: &[u8]) -> Self {
```

Comment:
It's awkward that this takes a slice when the newly-public `Header::from_bytes` takes a reference to an array. Let's make them consistent. Despite the breakage, on balance I favor making them both take arrays for explicitness.